nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/tools/command/PositionCommand.java

```diff
@@ -0,0 +1,150 @@
+                .longOpt("reset")
+                .desc("Resets all committed positions for the group")
+                .build());
+        options.addOption(Option.builder()
+                .longOpt("to-end")
+                .desc("Sets the current position to the end for all partitions")
+                .build());
+        options.addOption(Option.builder()
+                .longOpt("to-timestamp")
+                .desc("Sets the current position to a record >= timestamp, eg. " + Instant.now()
+                + " The timestamp used depends on the implementation, for Kafka this is the LogAppendTime. "
+                + "If no record offset is found with an appropriate timestamp then the command fails.")
+                .hasArg()
+                .argName("TIMESTAMP")
+                .build());
+    }
+
+    @Override
+    public boolean run(LogManager manager, CommandLine cmd) throws InterruptedException {
+        String name = cmd.getOptionValue("log-name");
+        String group = cmd.getOptionValue("group", "tools");
+
+        if (cmd.hasOption("to-timestamp")) {
+            long timestamp = parseTimestamp(cmd.getOptionValue("to-timestamp"));
+            if (timestamp < 0) {
+                return false;
+            }
+            return positionToTimestamp(manager, group, name, timestamp);
+        } else if (cmd.hasOption("to-end")) {
+            toEnd(manager,group,name);
+        } else if (cmd.hasOption("reset")) {
+            reset(manager,group,name);
+        } else {
+            System.err.println("Invalid option, try 'help position'");
+            return false;
+        }
+
+        return true;
+    }
+
+    protected void toEnd(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.toEnd();
+        }
+        System.out.println(String.format("# Moved log %s, group: %s, from: %s to %s", name, group, lag.lag(), lag.upper()));
+    }
+
+    protected void reset(LogManager manager, String group, String name) {
+        LogLag lag = manager.getLag(name, group);
+        long pos = lag.lower();
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            tailer.reset();
+        }
+        System.out.println(String.format("# Reset log %s, group: %s, from: %s to 0", name, group, pos));
+    }
+
+    protected boolean positionToTimestamp(LogManager manager, String group, String name, long timestamp) {
+
+        try (LogTailer<Record> tailer = manager.createTailer(group, name)) {
+            int size = manager.getAppender(name).size();
+            for (int partition = 0; partition < size; partition++) {
+                LogOffset logOffset = tailer.offsetForTimestamp(new LogPartition(name, partition), timestamp);
+                if (logOffset != null) {
+                    tailer.seek(logOffset);
+                    tailer.commit();
```

Comment:
here you are committing positions for all partitions, you should tailer.seek(logOffset) and
after the loop do a tailer.commit() only if you have updated some positions